pilot/pkg/config/kube/gateway/context.go

```diff
@@ -40,17 +42,23 @@ func NewGatewayContext(ps *model.PushContext) GatewayContext {
 // Note: this function considers *all* instances of the service; its possible those instances will not actually be properly functioning
 // gateways, so this is not 100% accurate, but sufficient to expose intent to users.
 // The actual configuration generation is done on a per-workload basis and will get the exact set of matched instances for that workload.
-// Three sets are exposed:
-// * Internal addresses (ie istio-ingressgateway.istio-system.svc.cluster.local:80).
-// * External addresses (ie 1.2.3.4), this comes from LoadBalancer services. There may be multiple in some cases (especially multi cluster).
+// Four sets are exposed:
+// * Internal addresses (eg istio-ingressgateway.istio-system.svc.cluster.local:80).
+// * External addresses (eg 1.2.3.4), this comes from LoadBalancer services. There may be multiple in some cases (especially multi cluster).
+// * Pending addresses (eg istio-ingressgateway.istio-system.svc), are LoadBalancer-type services with pending external addresses.
 // * Warnings for references that could not be resolved. These are intended to be user facing.
-func (gc GatewayContext) ResolveGatewayInstances(namespace string, gwsvcs []string, servers []*networking.Server) (internal, external, warns []string) {
+func (gc GatewayContext) ResolveGatewayInstances(
+	namespace string,
+	gwsvcs []string,
+	servers []*networking.Server,
+) (internal, external, pending, warns []string) {
```

Comment:
super nit: once we are at 4 args its hard to follow, maybe we can move it into a struct